File: AlgorithmReview/Programmers/Lv2/의상.swift
//
//  의상.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2023/05/02.
//

import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var map = [String: Int]()
    for i in clothes{
        if map[i[1]] == nil {
            map[i[1]] = 0
        }
        map[i[1]]! += 1
    }
    
    return map.values.reduce(1){ $0 * ($1 + 1) } - 1
}

File: Python/의상.py
def solution(clothes):
    
    map = {}
    for i in clothes:
        product, type = i[0], i[1]
        if type in map:
            map[type].append(product)
        else:
            map[type] = [product]
    
    result = 1
    for x in map.values():
        result *= len(x)+1
    
    return result - 1
